x-pack/packages/kbn-elastic-assistant/impl/data_anonymization_editor/helpers/index.ts

```diff
@@ -65,71 +33,45 @@ export const updateSelectedPromptContext = ({
     | 'deny'
     | 'denyReplacement';
 }): SelectedPromptContext => {
-  const { allow, allowReplacement } = selectedPromptContext;
+  const { contextAnonymizationFields } = selectedPromptContext;
+  if (!contextAnonymizationFields) {
+    return selectedPromptContext;
+  }
 
   switch (update) {
     case 'allow':
       return {
         ...selectedPromptContext,
-        allow: updateList({ field, list: allow, operation }),
+        contextAnonymizationFields: {
+          ...contextAnonymizationFields,
+          data: [
+            ...contextAnonymizationFields.data.filter((f) => f.field !== field),
+
+            {
+              // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+              ...contextAnonymizationFields.data.find((f) => f.field === field)!,
+              allowed: operation === 'add',
+            },
+          ],
+        },
       };
     case 'allowReplacement':
       return {
         ...selectedPromptContext,
-        allowReplacement: updateList({ field, list: allowReplacement, operation }),
+        contextAnonymizationFields: {
+          ...contextAnonymizationFields,
+          data: [
+            ...contextAnonymizationFields.data.filter((f) => f.field !== field),
+
+            {
+              // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+              ...contextAnonymizationFields.data.find((f) => f.field === field)!,
```

Comment:
```suggestion
              ...(contextAnonymizationFields.data.find((f) => f.field === field) ?? {
                id: '',
                field: '',
              }),
```

Again, I'm not sure if there is negative consequence for empty field/id but this will clear the type error. Same thing above on L51-52